### DISCUSSION TITLE
Would it be possible to create my desktop environment as XFCE, Gnome...etc. using fyne?

### DISCUSSION THREAD

**Discussion by ghost** [fyne-io/fyne#3251] [Q&A] [answered]
Hi all! 

I'm researching some easy-to-maintain or easy-to-prototype open source technology or library to create a design concept that would be my desktop environment. In this sense, I would like to know if fyne has technical resources to achieve this goal, which is to build a graphical environment in linux different from XFCE etc. 

Would it be possible to create my desktop environment as XFCE, Gnome...etc. using fyne?

**Comment by Bluebugs** [comment, marked as answer]
You might want to look at https://github.com/fyne-io/fynedesk . Short answer, yes!

**Reply by ghost** [reply]
Hi Bluebugs. After this answer, I am happy... But with a series of doubts(If you have time to help me, I would be glad for your help, feedback):
1. Is the memory performance good? What I mean, does it spend a lot of ram processing? 
2. Is there any analysis or study or any benchmarking comparing fyne-desktop vs kde or gnome  or XFCE to let me know if fyne-desktop is good for my specific use case? 
3. I'm reading the documentation and haven't found the information about benchmarking.
4. Is it possible to install fyne-desktop in node-os project or even Ubuntu?

**Reply by Bluebugs** [reply]
It depends what you are measuring, how you build things and what matters to you.

If you compare things to a highly optimize desktop that would share all library code and data across application at run time (think about texture, rendered font, rendered svg, ...), you will have a hard time to compete with Fyne. Now if you are looking at things that run a lot of C++ dependencies, do not do any of the above optimization, I think you will be in a good space for competing with them. Also I would expect as soon as you start a web browser, everything will be dwarfed by it.

Also currently fyne desktop doesn't have the complexity of something like KDE or GNOME, so it obviously doesn't have its runtime cost. Until it get to feature parity it will hard to compare more than just by being theorical. Maybe instead of looking at a benchmark, starting from the specification of how much resource from your computer you are ok spending and what features you are looking for, it would be possible to infer if that is doable.

**Reply by andydotxyz** [reply]
I run FyneDesk on a raspberry pi. It runs much better than Gnome or KDE alternatives but as @Bluebugs says that is in part due to fewer features

**Reply by ghost** [reply]
Hi andydotxyz. I would like to run fynedesk, but the OS doesn't have graphical support, I don't know where I could find information how to create a composer in node-os. _Do you know how to run or create a composer in node-os project?_

**Reply by andydotxyz** [reply]
It's not about a compositor, you need a graphics driver. Fyne already supports OpenGL and GLES. without one of those you'd need to build a new driver for Fyne that renders some other way.

**Reply by andydotxyz** [reply]
And currently you'd need X11 or Wayland to host the session, without that Fynedesk would need more bespoke code to host the session

**Reply by Bluebugs** [reply]
node-os is a linux os, but with user space being around node. You should be able to have mesa/x or mesa/wayland running on it and then run fyne desk just fine. Maybe what you want is go-os :-D

**Reply by ghost** [reply]
Hi Bluebugs and andydotxyz! That would be a great idea, although I don't have the knowledge for it. Anyway, thank you all so much for the feedback.
